spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -555,25 +555,39 @@ def has_seen_id_flag(obj)
           end
 
           def create_obj_in_recycled_slot(should_sample_original: false)
```

Comment:
Minor: In retrospect, I think it would be nice to have here a comment explaining what this bizarre method is trying to do, since it's really weird.

Reply:
Will add one!